Created Validator endpoints

DylanGTech opened the pull request:
I have created the APIs and tests for them as defined in the spec on the following issue
https://github.com/thenewboston-developers/dotnetcore-sdk/issues/20

One thing to note is that the models were inconsistent with the spec (more specifically, floating point decimal types were integers or strings). I took the time to change the models so that it was possible to create the tests and run them adequately and make them more consistent. It is definitely worth looking over before merging just in case, as there does appear to be confusion as to which decimal types (trust, the port numbers, and the values) are actually strings versus decimals.

Any changes that need to be made before a merge I am more than happy to make

xmac11 commented:
So far we've been matching the API responses, for example fetching the [config ](http://143.110.137.54/config)of a Bank return:
```
{
  ...
  "trust": "100.00"
  ...
}
```
You have spotted, however, some inconsistencies. For example, the port is indeed a nullable `int`.
What do you think @Mirch ? 
(I was actually expecting calling the API to fail after setting trust as a `decimal`, but the deserialization seems to convert it automatically)

joshman1019 commented:
It looks like the serialization process converts values to supported primitive data types. We may need to watch for conversion issues. From my research, the following conversion takes place: 


.NET | JSON
-- | --
String | String
ByteSByteUInt16Int16UInt32Int32UInt64Int64 | Integer
FloatDoubleDecimal | Float
Enum | Integer (can be the enum value name with StringEnumConverter)
DateTime | String (Serializing Dates in JSON)
Byte[] | String (base 64 encoded)
Type | String (type name)
Guid | String
TypeConverter (convertible to String) | String

joshman1019 commented:
I may need to dig into the documentation a bit more, but am I correct that the conversion between a base-ten value and a base-two value would cause substantial precision errors. 

For example, a calculation that resulted in 0.1 (decimal) would convert to 0.1000000014901161 (float)

Mirch commented:
We shouldn't rely on implicit conversion. Normally, I would use `decimal` internally and convert to string explicitly when communicating with the APIs. However, we are not using `trust` internally for anything, we are just wrapping it, so I guess matching the API response everywhere makes sense here? 
What do you guys think?

xmac11 commented:
I agree. We should also keep on eye on what the server is expecting from the client. For example, `trust` is a `decimal` when [patching an account.](https://thenewboston.com/bank-api/accounts)

Mirch commented on src/Thenewboston/Validator/Api/ValidatorService.cs:
```diff
@@ -1,13 +1,14 @@
 ﻿using System;
```
Please rename this file to ValidatorsService (along with the interface)

joshman1019 commented:
Agreed all around. So it looks like we need to modify a couple of the models to match the expectation of the server. Bank Transaction is currently using decimal as the type for "Amount". However I noticed in the Bank_Blocks documentation, that the "txs->Amount" is indicated as float. This also occurs in several other places. I believe that any transmission of currency should take place as a string and then be translated by either the server or client. Floating point will just not work for this type of transaction.

Mirch reviewed (changes requested):
I won't pinpoint all the fields, but you changed a lot of property types. Some make sense, some don't - please look at the API documentation and change them accordingly. While implicit conversion will take care of it, we should not rely on it and we should be explicit about what data we are sending to the API.

DylanGTech commented:
Reverted my changed to the model types that I made, so we don't have to rely on implicit conversion. Also renamed the class I made so it fits the convention as you requested. Thanks for your patience on that one. I'm new to this 👍

xmac11 commented:
No worries @DylanGTech , we're all pretty new here. Appreciate the help!

P.S. Another pitfall that should be emphasized, is that any field that refers to monetary value / coins is actually a [whole number](https://github.com/thenewboston-developers/Validator/blob/fce365808ab70265d9aa3beb40bc08887b94107a/v1/accounts/models/account.py#L11) with a [max value](https://github.com/thenewboston-developers/thenewboston-python/blob/9748ea9836066492de497672ec53bb2b0f501842/src/thenewboston/constants/network.py#L23) of `281_474_976_710_656`. The documentation on thenewboston is outdated. So my advice would be to always check by making a request to a live server. You can find IPs within the Account Manager app.

joshman1019 commented:
@xmac11 with that info I feel more comfortable jumping in. With my limited time I didn't realize that the docs were outdated. So I have kept my PR contributions limited. It's no wonder that I felt like something was "off" somewhere. Thanks for that.

Mirch commented:
We should really create some integrations test, it would make things way easier. I'll create an issue for that.
Meanwhile, can someone please review the rest of this? I am a bit busy these days. @joshman1019 @xmac11

joshman1019 commented:
@Mirch I am currently enslaved by the day job, but as I have time I will try to go through it unless @xmac11 wants to take this one.

xmac11 commented:
Yea, integration tests would be a good idea.

We should also finalize the structure of the service classes in the other issue. That's why I was not picking up any new issues these last days, but I might do some more and fix them altogether afterwards. So take your time right now @Mirch since you're busy, let's not rush it.

@DylanGTech could you please resolve the conflicts? Since you tried to revert the changes, it _might_ be easier to copy your new classes somewhere, start fresh in a new branch, and paste them back in.
